src/ReaderScreen/utils/gutkaScript.js

```diff
@@ -173,15 +225,19 @@ ${listener}.addEventListener(
         setAutoScroll();
       }
     }
-      // Handle scroll to saved position
+      // Handle scroll to saved element or position
     if (message.hasOwnProperty("action") && message.action === "scrollToPosition") {
-      const positionValue = parseFloat(message.position);
-      if (positionValue > 0 && positionValue <= 1) {
-        const maxScrollHeight = document.body.scrollHeight - window.innerHeight;
-        if (maxScrollHeight > 0) {
-          const scrollY = maxScrollHeight * positionValue;
-          window.scrollTo(0, scrollY);
-          curPosition = positionValue;
+      // Try element ID first if provided
+      if (message.elementId) {
+        const element = document.getElementById(String(message.elementId));
+        if (element) {
+          element.scrollIntoView({
+            behavior: "auto",
+            block: "start",
+            inline: "nearest"
+          });
+          console.log("No Element Found");
```

Comment:
Position of this console log appears wrong, it's inside the if statement for `element`